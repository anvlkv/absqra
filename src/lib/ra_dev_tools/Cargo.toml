[package]
name = "ra_dev_tools"
version = "0.1.0"
authors = ["a.nvlkv <a.nvlkv@protonmail.ch>"]
edition = "2018"
build = "build.rs"

[build-dependencies]
build-deps = "0.1.4"
# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
insta = "0.16.0"
regex = "1.3.9"
example_files_macro = {path="./src/lib/example_files_macro"}
files_iterator = {path="./src/lib/files_iterator"}
