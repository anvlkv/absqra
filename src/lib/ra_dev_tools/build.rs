extern crate build_deps;

fn main() {
    // Enumerate files in sub-folder "examples/*", being relevant for the test-generation (as example)
    // If function returns with error, exit with error message.
    build_deps::rerun_if_changed_paths( "../../../examples/*/*" ).unwrap();

    // Adding the parent directory "examples" to the watch-list will capture new-files being added
    build_deps::rerun_if_changed_paths( "../../../examples/*" ).unwrap();
}