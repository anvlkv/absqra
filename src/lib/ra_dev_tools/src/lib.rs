pub extern crate insta;
pub use files_iterator::for_each_ra_example_file;

#[allow(unused_attributes)]
#[macro_use] pub use example_files_macro::make_example_tests;

#[cfg(test)]
mod test {
    use super::make_example_tests;

    #[make_example_tests]
    #[test]
    fn it_works_with(content: String, file_name: String) {
        println!("didn't panic with content: \n[{}] of file_name: {}", content, file_name);
    }
}