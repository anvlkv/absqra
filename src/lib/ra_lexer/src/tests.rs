use super::*;
use ra_dev_tools::insta::assert_json_snapshot;
use ra_dev_tools::make_example_tests;

#[make_example_tests]
#[test]
fn it_should_match_snapshots(contents: String, file_name: String) {
    let tokens: Vec<RaToken> = tokenize(&contents).map(|r| r.unwrap()).collect();
    assert_json_snapshot!(file_name, tokens)
}