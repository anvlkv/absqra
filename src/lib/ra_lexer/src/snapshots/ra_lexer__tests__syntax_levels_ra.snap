---
source: src/lib/ra_lexer/src/tests.rs
expression: tokens
---
[
  {
    "kind": {
      "Identifier": "first_level"
    },
    "position": [
      [
        1,
        1
      ],
      [
        2,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "first_level_child"
    },
    "position": [
      [
        2,
        5
      ],
      [
        3,
        8
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "first_level_child_child"
    },
    "position": [
      [
        3,
        9
      ],
      [
        4,
        4
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "first_level_2nd_child"
    },
    "position": [
      [
        4,
        5
      ],
      [
        5,
        8
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "first_level_2nd_child_child"
    },
    "position": [
      [
        5,
        9
      ],
      [
        6,
        0
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "another_first_level"
    },
    "position": [
      [
        6,
        1
      ],
      [
        7,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "another_first_level_child"
    },
    "position": [
      [
        7,
        5
      ],
      [
        8,
        8
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "another_first_level_child_child"
    },
    "position": [
      [
        8,
        9
      ],
      [
        9,
        0
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "L1"
    },
    "position": [
      [
        10,
        1
      ],
      [
        11,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "L2_1"
    },
    "position": [
      [
        11,
        5
      ],
      [
        12,
        8
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "L3_1"
    },
    "position": [
      [
        12,
        9
      ],
      [
        13,
        8
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "L3_2"
    },
    "position": [
      [
        13,
        9
      ],
      [
        14,
        4
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "L2_1"
    },
    "position": [
      [
        14,
        5
      ],
      [
        15,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "L2_2"
    },
    "position": [
      [
        15,
        5
      ],
      [
        16,
        8
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "L3_3"
    },
    "position": [
      [
        16,
        9
      ],
      [
        17,
        12
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "L4"
    },
    "position": [
      [
        17,
        13
      ],
      [
        18,
        16
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "Identifier": "L5"
    },
    "position": [
      [
        18,
        17
      ],
      [
        19,
        20
      ]
    ],
    "level": 5
  },
  {
    "kind": {
      "Identifier": "L6"
    },
    "position": [
      [
        19,
        21
      ],
      [
        19,
        22
      ]
    ],
    "level": 6
  }
]
